components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,354 @@
+            DeviceManagementDAOFactory.rollbackTransaction();
+            String msg = "Error while getting Devices in Organization";
+            log.error(msg, e);
+            throw new DeviceOrganizationException(msg, e);
+        } catch (TransactionManagementException e) {
+            log.error("Error occurred while initiating transaction", e);
+        } finally {
+            DeviceManagementDAOFactory.closeConnection();
+            return devicesInOrganization;
+        }
+    }
+
+    @Override
+    public List<DeviceOrganizationMetadataHolder> getChildrenByParentId(String parentId) throws DeviceOrganizationException {
+        List<DeviceOrganizationMetadataHolder> children = new ArrayList<>();
+        try {
+            DeviceManagementDAOFactory.beginTransaction();
+            children = deviceOrganizationDAOimpl.getChildrenByParentId(parentId);
+            DeviceManagementDAOFactory.commitTransaction();
+        } catch (DeviceOrganizationDAOException e) {
+            DeviceManagementDAOFactory.rollbackTransaction();
+            String msg = "Error while getting children of Device " + parentId + " in organization";
+            log.error(msg, e);
+            throw new DeviceOrganizationException(msg, e);
+        } catch (TransactionManagementException e) {
+            log.error("Error occurred while initiating transaction", e);
+        } finally {
+            DeviceManagementDAOFactory.closeConnection();
+            return children;
+        }
+    }
+
+    @Override
+    public int getDeviceIndexInArray(String deviceId) {
+        List<DeviceOrganizationMetadataHolder> tempDevicesInOrganization = new ArrayList<>();
+        try {
+            tempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization";
+            log.error(msg, e);
+        }
+        int indexInArray;
+        for (DeviceOrganizationMetadataHolder tempHolder : tempDevicesInOrganization) {
+            if (tempHolder.getDeviceId().equals(deviceId)) {
+                indexInArray = tempDevicesInOrganization.indexOf(tempHolder);
+                return indexInArray;
+            }
+        }
+        return -1;
+    }
+
+    @Override
+    public List<DeviceOrganizationVisNode> generateNodes() {
+        List<DeviceOrganizationMetadataHolder> tempDevicesInOrganization = new ArrayList<>();
+        try {
+            tempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization database";
+            log.error(msg, e);
+        }
+        List<DeviceOrganizationVisNode> nodes = new ArrayList<>();
+        int serverFlag = 0;
+        for (DeviceOrganizationMetadataHolder tempHolder : tempDevicesInOrganization) {
+            if (tempHolder.getDeviceId().equals(SERVER_ID)) {
```

Comment:
I think here also it is better if you could use Yoda Expression here. That is SERVER_ID.equals(tempHolder.getDeviceId()).

Reply:
Fixed